contracts/protocol/pool/PoolApeStaking.sol

```diff
@@ -120,20 +120,16 @@ contract PoolApeStaking is
     /// @inheritdoc IPoolApeStaking
     function withdrawBAKC(
         address nftAsset,
-        ApeCoinStaking.PairNftWithdrawWithAmount[] memory _nftPairs
+        ApeCoinStaking.PairNftWithdrawWithAmount[] calldata _nftPairs
     ) external nonReentrant {
         DataTypes.PoolStorage storage ps = poolStorage();
         checkSApeIsNotPaused(ps);
 
-        address xTokenAddress;
-        {
-            DataTypes.ReserveData storage nftReserve = ps._reserves[nftAsset];
-            xTokenAddress = nftReserve.xTokenAddress;
-        }
-
-        INTokenApeStaking nTokenApeStaking = INTokenApeStaking(xTokenAddress);
-        IERC721 bakcContract = nTokenApeStaking.getBAKC();
+        address xTokenAddress = ps._reserves[nftAsset].xTokenAddress;
+        IERC721 bakcContract = INTokenApeStaking(xTokenAddress).getBAKC();
+        address bakcNToken = ps._reserves[address(bakcContract)].xTokenAddress;
```

Comment:
bakcXTokenAddress and bakcToken are maybe better because they are token and address